### PR TITLE
HTML5 step attribute support

### PR DESCRIPTION
**Pull request by etylsarin**
lightweight update to support step attribute for numeric values.
Fixes issue #1295

### CODE REVIEW

**Comment by staabm** [comment]
please add a unit test supporting the feature.

the build is failing because of code-style issues.. please fix that.

File: src/core.js [hunk 1]
```diff
@@ -1232,6 +1233,11 @@ $.extend( $.validator, {
 		range: function( value, element, param ) {
 			return this.optional( element ) || ( value >= param[ 0 ] && value <= param[ 1 ] );
 		},
+		
+		// http://jqueryvalidation.org/step-method/
```
Comment (staabm) [inline src/core.js]: this would require a additional docs PR at https://github.com/jzaefferer/validation-content

**Comment by jzaefferer** [comment]
Thank you for the contribution! Since this is based on the HTML5 spec, I'm good with adding it to the core of the plugin. Otherwise, what @staabm said - style fixes, unit tests and docs PR would be necessary to land this.

**Comment by pierrocknroll** [comment]
Thanks. Can you merge this ?

**Comment by staabm** [comment]
The abkve mentioned issues are still not resolved

**Comment by Arkni** [comment]
@etylsarin 
any update ?

**Comment by etylsarin** [comment]
I've done the updates of UTs, unfortunately git marked files from demo/cinema/ as changed as well and I don't know what to do with that.

File: src/core.js [hunk 2]
```diff
@@ -1316,6 +1317,11 @@ $.extend( $.validator, {
 			return this.optional( element ) || ( value >= param[ 0 ] && value <= param[ 1 ] );
 		},
 
+		// http://jqueryvalidation.org/step-method/
```
Comment (staabm) [inline src/core.js]: the docs for this url need to be created by a separate PR in https://github.com/jzaefferer/validation-content

**Comment by staabm** [comment]
regarding demo/cinema. you changed the mode/permission-bits on those files.

**Comment by etylsarin** [comment]
@staabm - regarding demo/cinema, yeah, that happened during the merge, but I'm not sure now how to remove these files from the pull request. I've also created a separate pull request on docs.

**Comment by Arkni** [comment]
@etylsarin 
You should rebase your branch on top of `upstream/master` (this will give you some merge conflicts, that you should fix).

Also, you should prefix your first commit with `Core:` and add a new line before `Fixes #1295`

**Comment by Arkni** [comment]
That's a huge number of commits.
I suggest you to backup your changes and create a new and clean branch (`patch-1` as name) from master and then apply your changes, then commit and `git push -f origin patch-1` to overwrite this pull request.

**Comment by etylsarin** [comment]
Well, that's what happened when I'd done rebase.
Anyway, the only pending changes are related to the fix itself and the history of my branch isn't crucial, is it?

**Comment by Arkni** [comment]
The two lines to the right (red and pink) are a result of your merge/rebase.

anim

**Comment by Arkni** [comment]
@etylsarin 
Are you willing to rebase this PR ?

**Comment by etylsarin** [comment]
I have created a new PR #1638

**Comment by slievain** [comment]
Hello, any update?

**Comment by Arkni** [comment]
@slievain 
Merged in b3c07e39571a8e6a0583ef9a50ca10681374890e

**Comment by slievain** [comment]
Thanks! When will the next pre-build version be released?

**Comment by Arkni** [comment]
> When will the next pre-build version be released?

Really don't know, maybe @staabm have an approximation.

**Comment by staabm** [comment]
I am on vacation for the rest of the week. I hope to get a release tagged next week

**Comment by lanmower** [comment]
Perfect example of where developers 'just don't care' that their project is incomplete, and drag their feet every step of the way, providing no legwork for the improvement of their own projects.

Meanwhile this project has become useless over the course of the last 5 years because of this negligence.

For shame.

**Comment by Arkni** [comment]
@lanmower 

I don't know what you're talking about or what you want to point out in your comment. If you are talking about this PR, the OP opened another PR with the same content and it was merged 3 years ago.

Also, just a **Friendly advice** to you and to anyone else that have an opinion like yours:

_**Please do not ask us (or any other FOSS developers) to do something "today" or with other urgency. We have our daily job, family to feed, kids to take care of, volunteering tasks, and other wonderful life activities. Often times, we can't afford the time to worry about someone's else problems, not out of malicious intention but simply because there's only so many things one can help.**_

Have a nice day!
